### DISCUSSION TITLE
I need help about "themes.xml" in android studio

### DISCUSSION THREAD

**Discussion by win-luck** [euphony-io/euphony#154] [Q&A] [answered]
hello, I haven't dealt with android studio before, So I am studying it myself..
I know We sholud use Compose If we want to make app with android studio, 
but in the Empty Activity(not Compose), I have a question about "theme"

![image](https://user-images.githubusercontent.com/53044069/179512371-5862be7a-1e62-4def-bd5b-ef38ce448ecb.png)

![image](https://user-images.githubusercontent.com/53044069/179512239-b03cca95-a908-4028-b15e-b23100a07021.png)

My question is:
* What is the "themes.xml"?
* What is the difference in two "themes.xml"?

**Comment by designe** [comment]
Are you wondering about the difference between having only one `themes.xml` and having two in themes folder?

**Reply by win-luck** [reply]
right. I found many website about it, but I don't understand well.

**Comment by hyunjung-choi** [comment, marked as answer]
`themes.xml`manages app's resource such as color, style,,,ect. It affects the entire app.

Making `themes` folder is default.
But `theme.xml` makes one in the Jetpack compose project.
This reason finds yourself! :)
It'll be good to compose study to you!

## Reference
https://stackoverflow.com/questions/64352950/android-theme-xml-and-theme-xmlnight

**Reply by win-luck** [reply]
is it like a CSS?

**Reply by hyunjung-choi** [reply]
> is it like a CSS?

Hmm... nope!

**Reply by hyunjung-choi** [reply]
> is it like a CSS?

It includes not only style but also other Android resources. For example, there is a Material Components design in which the entire app is inherited.

**Reply by win-luck** [reply]
thank you for your help :)

**Comment by kldaji** [comment]
- What is the "themes.xml"?

Theme defines style of your application.
Style means kind of resource such as color, font, etc as @hyunjung-choi  mentioned.
In manifest file, you can find application tag which represent your application with `theme` attribute.
![스크린샷 2022-07-19 오전 8 19 36](https://user-images.githubusercontent.com/78070388/179632975-acb9ade5-5d1c-49cf-a32f-fcfbdd5947ea.png)

And I think you may wonder why we have to define `themes.xml`.
Actually, it is time consuming for developers to define resources from scratch, so Android provides a number of predefined theme to developer.
![스크린샷 2022-07-19 오전 8 26 55](https://user-images.githubusercontent.com/78070388/179633450-264a7c36-3db1-413c-8459-c35a1b355fa2.png)

So, we just inherit the predefined theme unless you want to custom your theme.
![스크린샷 2022-07-19 오전 8 27 58](https://user-images.githubusercontent.com/78070388/179633623-7cab4ac3-ff64-4008-b7d3-7d6e33209523.png)

And it may also helpful reference [Material Design](https://material.io/design/material-theming/implementing-your-theme.html#color) to understand what is `colorPrimary` stands for?

- What is the difference in two "themes.xml"?

Android application supports light theme and dark theme.
That's why there exists two `themes.xml`.
Basically, Android Studio provides two `themes.xml` as you already know, you can see that the color and font will be changed when you toggle to dark mode in your device.


I made an effort to make it easier to understand 😂, and I hope it helps. 👍👍

**Reply by win-luck** [reply]
I have to read official text.. thank you for you kindly help :)
